src/PrestaShopBundle/Entity/Mutation.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Entity;
+
+use DateTime;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * Mutation entity allows to track a modification/mutation performed on a table, usually in the BO, but it could
+ * also allow tracking some modification in FO like a payment module changing an order status.
+ *
+ * As such the mutation needs to hold three mandatory elements:
+ * - mutation table: which table was modified
+ * - mutation row ID: ID identifying which row of the modified table was modified
+ * - action performed (create, update, delete) identified by an enum to remain small in DB
+ * - mutator type: Employee, ApiClient or Module
+ * - mutator identifier: Identifier of associated mutator (usually an int matching the row, but can be a technical name for a module)
+ *
+ * @ORM\Table
+ * @ORM\HasLifecycleCallbacks
+ * @ORM\Entity(repositoryClass="PrestaShopBundle\Entity\Repository\MutationRepository")
+ */
+class Mutation
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(name="id_mutation", type="integer", options={"unsigned": true})
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private int $id;
```

Comment:
```suggestion
```

I think we agreed not to use `ID` for this table. Is there any usage for it?

Reply:
Yeah actually Doctrine doesn't allow entities without primary keys For regular associations we would use the couple of IDs as primary keys but the thing is that in this case we can't because the associations may be duplicate (same object same id over different mutations in time) So we need this independent ID, it's mandatory and a good practice to have this primary key especially in case we need to handle the DELETION in the future (we won't be able to delete based on associations as some may need to be removed and some not)

I don't know if it's clear 😅